cmd/client-s3.go

```diff
@@ -874,24 +884,48 @@ func (c *s3Client) Stat(isIncomplete bool) (*clientContent, *probe.Error) {
 		if objectStat.Err != nil {
 			return nil, probe.NewError(objectStat.Err)
 		}
-
 		if objectStat.Key == object {
 			objectMetadata.URL = *c.targetURL
 			objectMetadata.Time = objectStat.LastModified
 			objectMetadata.Size = objectStat.Size
 			objectMetadata.ETag = objectStat.ETag
 			objectMetadata.Type = os.FileMode(0664)
-			objectMetadata.Metadata = map[string][]string{}
+			objectMetadata.Metadata = map[string]string{}
+			objectMetadata.EncryptionHeaders = map[string]string{}
+			if isFetchMeta {
+				stat, err := c.getObjectStat(bucket, object)
+				if err != nil {
+					return nil, err
+				}
+				objectMetadata.Metadata = stat.Metadata
+				objectMetadata.EncryptionHeaders = stat.EncryptionHeaders
+			}
 			return objectMetadata, nil
 		}
 
 		if strings.HasSuffix(objectStat.Key, string(c.targetURL.Separator)) {
 			objectMetadata.URL = *c.targetURL
 			objectMetadata.Type = os.ModeDir
-			objectMetadata.Metadata = map[string][]string{}
```

Comment:
This is a directory (os.ModeDir), I don't think we need to fetch for Metadata here.